### PR TITLE
Samples and tests updated with relaxed final constraint for primary keys and properties

### PR DESCRIPTION
**Pull request by desistefanova**
Removed `final` from properties, primary keys and lists in realm models.
Realm objects are regenerated.
All the tests and samples are tested with the new models.

### CODE REVIEW

File: test/realm_test.dart [hunk 1]
```diff
@@ -400,11 +400,9 @@ Future<void> main([List<String>? args]) async {
 
       expect(car.make, equals('Tesla'));
 
-      expect(() {
-        realm.write(() {
-          car.make = "Audi"; // setting late final is a runtime error :-/
-        });
-      }, throws<RealmUnsupportedSetError>());
+      realm.write(() {
```
Comment (blagoev) [inline test/realm_test.dart:403]: this one should still fail

**Review by blagoev — CHANGES_REQUESTED** [review]

File: test/realm_test.dart (hunk 1 above)
Comment (blagoev) [reply test/realm_test.dart:403]: this one fixes this https://github.com/realm/realm-dart/pull/253 consider @desistefanova could you please revert this test change

Comment (blagoev) [reply test/realm_test.dart:403]: I guess we need to move #253 first and then rebase this one so we are sure the tests behaves the same

Comment (desistefanova) [reply test/realm_test.dart]: > this one should still fail

Expected exception is reverted with https://github.com/realm/realm-dart/pull/248/commits/8c240e43b05bee6712223a5588ff8e65d4e02ac6.

Comment (desistefanova) [reply test/realm_test.dart]: > I guess we need to move #253 first and then rebase this one so we are sure the tests behaves the same

I did rebase onto #253 and regenerated all the models again. 
I removed `final` from Lists also.

File: .github/workflows/realm-dart-linux.yml [hunk 2]
```diff
@@ -37,14 +37,10 @@ jobs:
...
 
-      # Note: This workflow uses the latest stable version of the Dart SDK.
-      # You can specify other versions if desired, see documentation here:
-      # https://github.com/dart-lang/setup-dart/blob/main/README.md
-      # - uses: dart-lang/setup-dart@v1
       - name : Setup Dart SDK
         uses: dart-lang/setup-dart@main
         with:
-          sdk: beta
+          sdk: stable
```
Comment (blagoev) [inline .github/workflows/realm-dart-linux.yml:43]: `realm-dart-linux.yaml` and `realm-generator.yaml` already contain the changes, so we should not change it here. Could you please remove these files from the PR

File: generator/lib/src/field_element_ex.dart [hunk 3]
```diff
@@ -185,15 +185,6 @@ extension FieldElementEx on FieldElement {
       } else {
         // Validate collections
         if (type.isRealmCollection) {
-          if (!isFinal) {
```
Comment (blagoev) [inline generator/lib/src/field_element_ex.dart:188]: These changes are part of master from #253 Could you please remove these files changes from this PR. So all generator files changes here

File: test/realm_test.dart [hunk 4]
```diff
@@ -402,7 +402,7 @@ Future<void> main([List<String>? args]) async {
 
       expect(() {
         realm.write(() {
-          car.make = "Audi"; // setting late final is a runtime error :-/
+          car.make = "Audi"; // setting primary key is a runtime error :-/
```
Comment (blagoev) [inline test/realm_test.dart:405]:
```suggestion
          car.make = "Audi";
```

**Review by blagoev — CHANGES_REQUESTED** [review]

File: .github/workflows/realm-dart-linux.yml (hunk 2 above)
Comment (blagoev) [reply .github/workflows/realm-dart-linux.yml:43]: this one is in master now. ignore it here

Comment (desistefanova) [reply .github/workflows/realm-dart-linux.yml]: Done. Merged from `master.`

File: generator/lib/src/field_element_ex.dart (hunk 3 above)
Comment (desistefanova) [reply generator/lib/src/field_element_ex.dart]: Done. Merged from `master`.

**Review by blagoev — APPROVED** [review]

**Pull request merged**